version/version.go

```diff
@@ -64,6 +73,41 @@ func (v *Version) AtLeast(minVersion string) bool {
 	return v.Compare(minVersion) <= 0
 }
 
+// Check if a candidate is an upgrade/downgrade of the base version only in the specified position
+// Which corresponds to Major, Minor and Patch positions.
+// Examples: currVersion = Version{"1.2.1}
+// currVersion.compareAtPosition(Version{"1.2.2"}, Patch) => 1.
+// currVersion.compareAtPosition(Version{"1.2.0"}, Patch) => -1.
+// currVersion.compareAtPosition(Version{"1.2.1"}, Major) => 0.
+func (v *Version) CompareAtPosition(candidate Version, versionPosition VersionPosition) (int, error) {
+	v1Parts := strings.Split(strings.Trim(v.version, "v"), ".")
+	v2Parts := strings.Split(strings.Trim(candidate.version, "v"), ".")
+	for i := 0; i < 3; i++ {
```

Comment:
What do we need to iterate over all the positions,. if we have one position only we need to compare? Can't we "jump" to the index of the position we need to compare?